geomstats/learning/k_means.py

```diff
@@ -0,0 +1,87 @@
+import random
+
+import geomstats.backend as gs
+from geomstats.learning._template import TemplateTransformer
+
+
+class KMeans(TemplateTransformer):
+
+    def __init__(self, n_clusters, metric, init="random",
+                 tol=1e-2, verbose=0):
+        """ K-Means algorithm using Riemannian manifolds
+
+        Parameters
+        ----------
+        n_clusters : Number of clusters (k value of the k-means)
+
+        metric : The geomstats metric associate to the space used
+
+        init : How to init centroids at the beginning of the algorithm.
+               Only radom is implemented yet.
+
+        tol : convergence factor. If the difference of mean distance
+             between two step is lower than tol
+
+        verbose : if verbose > 0, information will be print during learning
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        self.n_clusters = n_clusters
+        self.init = init
+        self.metric = metric
+        self.n_clusters = n_clusters
+        self.tol = tol
+        self.verbose = verbose
+
+    def fit(self, X, max_iter=100):
+        """Predict for each data point the closest center in terms of metric distance
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        max_iter : Maximum number of iteration
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        belongs = gs.zeros(X.shape[0])
+        self.centroids = gs.concatenate([gs.expand_dims(X[
+                                         random.randint(0, X.shape[0]-1)], 0)
```

Comment:
Is there a reason why you use random.randint and not some numpy function?

Reply:
I would also do something like:
```
n_points = X.shape[0]
random_indices = gs.random.randint(n_points, size=self.n_clusters)
self.centroids = X[random_indices]
```
if it runs? It can make it shorter, more readable and factorize X.shape[0] into n_points that you use twice.